staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity.go

```diff
@@ -213,11 +202,11 @@ func (t *nodeSelectorTerm) match(nodeLabels labels.Set, nodeFields fields.Set) (
 
 // nodeSelectorRequirementsAsSelector converts the []NodeSelectorRequirement api type into a struct that implements
 // labels.Selector.
-func nodeSelectorRequirementsAsSelector(nsm []v1.NodeSelectorRequirement, path *field.Path) (labels.Selector, field.ErrorList) {
+func nodeSelectorRequirementsAsSelector(nsm []v1.NodeSelectorRequirement, path *field.Path) (labels.Selector, error) {
```

Comment:
I prefer this returns `[]error`

In general, whatever is non exported could return `[]error`, but whatever is exported should return `Flatten`.

Make sure `LazyErrorNodeSelector.Match` flattens the errors.

Reply:
done